x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/execution/ExecutionService.java

```diff
@@ -106,7 +105,7 @@
     private final WatchExecutor executor;
     private final ExecutorService genericExecutor;
 
-    private AtomicReference<CurrentExecutions> currentExecutions = new AtomicReference<>();
+    private CurrentExecutions currentExecutions;
```

Comment:
Can `CurrentExecutions` remain inside an `AtomicReference`? 
It is read else where without acquiring a lock and as far as I understand it is about making sure that `clearExecutions()` happens in a sync manner, which is possible with keeping the `AtomicReference`?

(also the currentExecutions field can then be made final)

Reply:
Yes, I think this either needs to be `volatile` or go back to being an `AtomicReference`.

Reply:
thanks nice catch. I also had messed up the order of "sealing" the concurrent executions.  The change here is now a single line that removes the fork.